src/core/net/ip6.cpp

```diff
@@ -73,6 +73,31 @@ Message *Ip6::NewMessage(uint16_t aReserved, const otMessageSettings *aSettings)
         Message::kTypeIp6, sizeof(Header) + sizeof(HopByHopHeader) + sizeof(OptionMpl) + aReserved, aSettings);
 }
 
+Message *Ip6::NewMessage(const uint8_t *aData, uint16_t aDataLength, const otMessageSettings *aSettings)
+{
+    otMessageSettings settings = {true, OT_MESSAGE_PRIORITY_NORMAL};
+    Message *         message;
+
+    if (aSettings != NULL)
+    {
+        settings = *aSettings;
+    }
+
+    VerifyOrExit(GetDatagramPriority(aData, aDataLength, *reinterpret_cast<uint8_t *>(&settings.mPriority)) ==
```

Comment:
nit: how about initializing `message` to `NULL` at declaration above, then using `SuccessOrExit()` here.